read.go

```diff
@@ -380,6 +397,15 @@ func NewSearcher(r IndexFile) (Searcher, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if sourcegraphInMemoryContent {
+		cached := &cachedIndexFile{file: r}
```

Comment:
I think since we're persisting `r` and it is an mmap'd file it means that we will pay the cost both of this in-memory cache AND the mmap'd file being paged into memory so this will appear as 2x the memory cost, I think? would be curious to hear if I am wrong about this.

If I am right, we could fix this easily using that commit I made https://github.com/sourcegraph/zoekt/pull/37/commits/5f6e1205f747a2d452b1e72d8fdaedb6c7a5efe6 basically

Reply:
> we could fix this easily using that commit I made 5f6e120 basically

I was wrong about this, `indexFileFromOS` tries to keep one file handle open per shard which ends up being a lot.